internal/keepers/ocr.go

```diff
@@ -52,15 +52,24 @@ func (k *keepers) Observation(ctx context.Context, rt types.ReportTimestamp, _ t
 		return nil, fmt.Errorf("%w: failed to sample upkeeps for observation: %s", err, lCtx)
 	}
 
+	// keyList produces a sorted result so the following reduction of keys
+	// should be more uniform for all nodes
 	keys := keyList(filterUpkeeps(results, ktypes.Eligible))
 
 	// limit the number of keys that can be added to an observation
 	// OCR observation limit is set to 1_000 bytes so this should be under the
 	// limit
-	if len(keys) > 10 {
-		keys = keys[:10]
+	var tot int
+	var idx int
+	// if the total plus padding for json encoding is less than the max, another
+	// key can be included
+	for tot+(5*(idx+1)) < maxObservationLength && idx < len(keys) {
+		tot += len(keys[idx])
+		idx++
```

Comment:
I don't fully understand the logic (especially the magical number 5). 
imo a constant limit is much simpler and the dynamic limit doesn't give much advantage, but will leave it upto you

Reply:
I'm not usually keen on magic number either, but it's based on the number of extra bytes for each key that would be added in the encode process.